src/pyggdrasil/scripts/cell_simulation.py

```diff
@@ -0,0 +1,369 @@
+    # Attach Cells To Tree
+    ###############################################################################
+    # convert adjacency matrix to self-connected tree - in tree_inference format
+    np.fill_diagonal(tree, 1)
+    # define strategy
+    strategy = sim.CellAttachmentStrategy[strategy]
+    # attach cells to tree - generate perfect mutation matrix
+    perfect_mutation_mat = sim.attach_cells_to_tree(
+        rng_cell_attachment, tree, n_cells, strategy
+    )
+
+    ###############################################################################
+    # Add Noise
+    ################################################################################
+    # add noise to perfect mutation matrix
+    noisy_mutation_mat = None
+    if (beta > 0) or (alpha > 0) or (na_rate > 0):
+        noisy_mutation_mat = sim.add_noise_to_perfect_matrix(
+            rng_noise, perfect_mutation_mat, alpha, beta, na_rate, observe_homozygous
+        )
+
+    ################################################################################
+    # Save Simulation Results
+    ################################################################################
+    # make save name and path from parameters
+    filename = compose_save_name(params, tree_no=tree_no) + ".json"
+    fullpath = os.path.join(outdir, filename)
+    # make output directory if it doesn't exist
+    os.makedirs(outdir, exist_ok=True)
+
+    # format tree for saving
+    root = adjacency_to_root_dfs(tree)
+    root_serialized = serialize_tree_to_dict(root, serialize_data=dummy_serialize)
+
+    # print tree if prompted verbose
+    if verbose:
+        print(root)
+
+    # Save the data to a JSON file
+    # Create a dictionary to hold matrices
+    if noisy_mutation_mat is not None:
+        data = {
+            "adjacency_matrix": tree.tolist(),
+            "perfect_mutation_mat": perfect_mutation_mat.tolist(),
+            "noisy_mutation_mat": noisy_mutation_mat.tolist(),
+            "tree": tree.tolist(),
+            "root": root_serialized,
+        }
+    else:
+        data = {
+            "adjacency_matrix": tree.tolist(),
+            "perfect_mutation_mat": perfect_mutation_mat,
+            "tree": tree.tolist(),
+            "root": root_serialized,
+        }
+
+    # Save the data to a JSON file
+    with open(fullpath, "w") as f:
+        json.dump(data, f)
+
+    # Print the path to the file if verbose
+    if verbose:
+        print(f"Saved simulation results to {fullpath}\n")
+
+
+def run_sim(params) -> None:
+    """Generate {n_trees} of simulated data and save to disk.
+
+    Args:
+        params: dict
+            input parameters from parser for simulation
+
+    Returns:
+        None
+    """
+
+    # Create a random number generator
+    rng = random.PRNGKey(params["seed"])
+
+    # Generate n_simulations of simulated data
+    for i in range(params["n_trees"]):
+        print(f"Generating simulation {i+1}/{params['n_trees']}")
```

Comment:
Currently `rng` is used twice. The JAX way of doing it is splitting the key before the usage.

In other words, this loop should read:
```python
 for i in range(params["n_trees"]):
        print(f"Generating simulation {i+1}/{params['n_trees']}")
         rng, key_iter = random.split(rng)
        gen_sim_data(params, key_iter, tree_no=i + 1)
```

Reply:
There's also another way, which I actually like the most, with generating different random keys for different iterations at the start:

```python
# Create a random number generator
rng = random.PRNGKey(params["seed"])
keys = random.split(rng, params["n_trees"])
for i, key in enumerate(keys, 1):
    print(f"Generating simulation {i}/{len(keys)}")
    gen_sim_data(params, key, tree_no=i)
```

Reply:
Oh, yes -  good catch -  implemented your second proposal.